ssa/manager_apply.go

```diff
@@ -201,3 +214,53 @@ func (m *ResourceManager) apply(ctx context.Context, object *unstructured.Unstru
 	}
 	return m.client.Patch(ctx, object, client.Apply, opts...)
 }
+
+const kubectlManager = "kubectl"
+
+// cleanupManagedFields removes the client-side-apply managers and kubectl server-side-apply
+// from metadata.managedFields and patches the object in-cluster.
+// Workaround for Kubernetes bug https://github.com/kubernetes/kubernetes/issues/99003.
+func (m *ResourceManager) cleanupManagedFields(ctx context.Context, object *unstructured.Unstructured) error {
+	if object == nil {
+		return nil
+	}
+
+	existingObject := object.DeepCopy()
+	existingFields := existingObject.GetManagedFields()
+	if len(existingFields) == 0 {
+		return nil
+	}
+
+	var fields []metav1.ManagedFieldsEntry
```

Comment:
we could replace this by `fields := make([]metav1.ManagedFieldsEntry, 0, len(existingFields))`, to avoid extra allocation in the append.

Reply:
Have you tried it? Looks to me that it breaks the whole thing.

Reply:
Yes one of the tests fails
```
manager_apply_test.go:487: Mismatch from expected value (-want +got):
              string(
            -   "resource-manager",
            +   "before-first-apply",
              )
```

I fail to see why for now...